### PR TITLE
Dockerize

### LINKED ISSUE #179: Dockerize
We want to make it easier to test and/or contribute to this repo utilizing Docker. This has been completed in [Python](https://github.com/sendgrid/sendgrid-python/tree/master/docker), please use that example as a reference.

Key features:
* Setup the environment as specified in the CONTRIBUTING.md file
* Prism should be running on launch
* The README and USAGE file should help the user get up and running quickly

### PR DESCRIPTION
**Pull request by shrivara**
This covers the basic cases of working on the gem and running tests. I just wanted to get a quick review to understand what else is required.

https://github.com/sendgrid/sendgrid-ruby/issues/179

@mbernier edit: resolves #179

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=181) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=181) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-ruby?pullRequest=181) it.</sub>

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Docker image for sendgrid-ruby
+
+## Quickstart
+1. [Install Docker](https://docs.docker.com/engine/installation/) on your machine.
+2. Run `docker run --rm -it sendgrid/sendgrid-ruby irb`.
```
Comment (mbernier) [inline docker/README.md]: What is the `irb` at the end there?

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+## Poke around
+
+If you would like to just poke around in the image and check some examples:
+```sh
+docker run --rm -it sendgrid/sendgrid-ruby bash
+```
+
+If you want to mount your fork or specific version of the gem:
+```sh
+docker run --rm -v /path/to/local/sendgrid-ruby:/sources/sendgrid-ruby -it sendgrid/sendgrid-ruby bash
```
Comment (mbernier) [inline docker/README.md]: Would you be up for adding a docker-compose file that would make the ruby gems and source code local to the machine the container is running on?

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+## Running tests
```
Comment (mbernier) [inline docker/README.md]: This is awesome. thank you!!

**Comment by mbernier** [comment]
@shrivara

We have not been able to merge your Pull Request, but because you are awesome - we wanted to make sure you could still get a SendGrid Hacktoberfest shirt.  

Please go **fill [out our swag form](https://dx.sendgrid.com/swag) before Nov 5th** and we will send the shirt! (We know that you might have tried this before and it didn’t work, sorry about that!)

### **You have till Nov 5th to fill out this form** in order to get the Hacktoberfest shirt!

Thank you for contributing during Hacktoberfest! We hope to see you in the repos soon! Just so you know, we always give away a SendGrid shirt for your first ever non-Hacktoberfest PR that gets merged.

File: docker/README.md (hunk 1 above)
Comment (shrivara) [reply docker/README.md]: The idea was to get a interpretor where a dev could try out the gem without having to run it.

File: docker/README.md (hunk 2 above)
Comment (shrivara) [reply docker/README.md]: Will the volumes not be sufficient for this?

**Comment by thinkingserious** [comment]
Hello @shrivara,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Comment by mbernier** [comment]
If you are interested, we would love the opportunity to talk to you about Hacktoberfest and our API libraries.

Our agenda would be: Explore what you liked and is there anything we can do to improve? 

You can grab a time [on my calendar](https://calendly.com/matt-bernier/30min) that works for you and we can have a chat on Google Hangout or Skype. If you prefer, you can email me using my GitHub username at my company’s domain.

Thank you so much,

Matt Bernier - @mbernier - SendGrid Developer Experience Product Manager
Elmer Thomas - @thinkingserious - SendGrid Developer Experience Engineer
@SendGrid

**Pull request merged**